index.js

```diff
@@ -290,60 +361,64 @@ function ensureSlab (size) {
   return SLAB
 }
 
-function encodeIndexRange (prefix, opts) {
+function encodeIndexRange (pointer, type, opts) {
   const bounded = { gt: null, gte: null, lte: null, lt: null, reverse: !!opts.reverse, limit: opts.limit || Infinity }
 
-  if (opts.gt || opts.gt === 0) bounded.gt = encodeIndex(prefix, opts.gt)
-  else if (opts.gte) bounded.gte = encodeIndex(prefix, opts.gte)
-  else bounded.gte = encodeIndex(prefix, 0)
+  if (opts.gt || opts.gt === 0) bounded.gt = encodeDataIndex(pointer, type, opts.gt)
+  else if (opts.gte) bounded.gte = encodeDataIndex(pointer, type, opts.gte)
+  else bounded.gte = encodeDataIndex(pointer, type, 0)
 
-  if (opts.lt || opts.lt === 0) bounded.lt = encodeIndex(prefix, opts.lt)
-  else if (opts.lte) bounded.lte = encodeIndex(prefix, opts.lte)
-  else bounded.lte = Buffer.concat([prefix, INF]) // infinity
+  if (opts.lt || opts.lt === 0) bounded.lt = encodeDataIndex(pointer, type, opts.lt)
+  else if (opts.lte) bounded.lte = encodeDataIndex(pointer, type, opts.lte)
+  else bounded.lte = encodeDataIndex(pointer, type, Infinity) // infinity
 
   return bounded
 }
 
-function encodeBatchPrefix (prefix, batchId, type) {
-  const state = ensureSlab(128)
-  const start = state.start
-  state.buffer.set(prefix, start)
-  state.start += prefix.byteLength
-  UINT.encode(state, batchId)
-  UINT.encode(state, type)
-  return state.buffer.subarray(start, state.start)
+function encode (encoding, value) {
+  let slab = 128
+  while (slab <= SLAB.end) {
```

Comment:
you dont need the while here, the slab, ie 128, should just be set to something that is guaranteed to be bigger than any key which i think 128 is. otherwise encoding will throw also